rwengine/src/ai/CharacterControllerActivities.hpp

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+#ifndef _CONTROLLER_ACTIVITY_HPP_
+#define _CONTROLLER_ACTIVITY_HPP_
+
+#include <ai/CharacterController.hpp>
+
+
+#define DECL_ACTIVITY(act_name)                                               \
+	class act_name : public CharacterController::Activity                 \
+	{                                                                     \
+	public:                                                               \
+		static constexpr auto ActivityName = #act_name;               \
+		std::string name() const override { return ActivityName; };   \
+		bool update(CharacterObject*, CharacterController*) override;
+
+#define DECL_ACTIVITY_CANSKIP(act_name)                          \
+	DECL_ACTIVITY(act_name)                                  \
+		bool canSkip(CharacterObject *) const override;
```

Comment:
Strings and templates don't cooperate very well.

What I have in mind is replacing the string names with a tagged union or variant so we don't allocate an object for each activity.